src/Peachpie.CodeAnalysis/Utilities/DistinctQueue.cs

```diff
@@ -44,10 +50,10 @@ public bool Enqueue(T value)
 
             lock (_syncRoot)
             {
-                if (!value.IsEnqueued)
+                if (!_set.Contains(value))
```

Comment:
`_set.Contains(x) and _set.Add(x)` can be simplified to single lookup `!_set.Add(x)`

Reply:
OK, fixed in 7d635e6.